src/chrome/internal/scripts/script.ts

```diff
@@ -0,0 +1,117 @@
+import * as fs from 'fs';
+import {
+    ILoadedSource, SourceOfCompiled, ScriptRunFromLocalStorage, DynamicScript,
+    ScriptRuntimeSource, ScriptDevelopmentSource, NoURLScriptSource
+} from '../sources/loadedSource';
+import { CDTPScriptUrl } from '../sources/resourceIdentifierSubtypes';
+import { IValidatedMap } from '../../collections/validatedMap';
+import { printArray } from '../../collections/printing';
+import { ISourcesMapper } from './sourcesMapper';
+import { IResourceIdentifier, IResourceLocation, newResourceIdentifierMap, parseResourceIdentifier, ResourceName } from '../sources/resourceIdentifier';
+import { IExecutionContext } from './executionContext';
+
+/** This interface represents a piece of code that is being executed in the debugee. Usually a script matches to a file or a url, but that is not always the case.
+ * This interface solves the problem of finding the different loaded sources associated with a script, and being able to identify and compare both scripts and sources easily.
+ */
+export interface IScript {
+    readonly executionContext: IExecutionContext;
+    readonly runtimeSource: ILoadedSource<CDTPScriptUrl>; // Source in Webserver
+    readonly developmentSource: ILoadedSource; // Source in Workspace
+    readonly sourcesOfCompiled: SourceOfCompiled[]; // Sources before compilation
```

Comment:
What's the difference between a development source and a "source of compiled"? Aren't the development sources the ones that are compiled?

Reply:
A developmentSource is the .js location inside the user workspace
source of compiled now renamed to mapped source is the sources that we get out of the SourceMap such as typescript sources.